src/firmware/main/math/polynomial_2d.c

```diff
@@ -0,0 +1,178 @@
+        float initial_value        = calculateL2NormAtValueOrder##N_minus_1(deriv, t_min); \
+        float sum_of_odd_s_values  = 0;                                                    \
+        float sum_of_even_s_values = 0;                                                    \
+        parametrization.s_values[0] = 0;                                                   \
+        parametrization.t_values[0] = t_min;                                               \
+        for (size_t i = 1; i < parametrization.num_values; i++)                            \
+        {                                                                                  \
+            const float t_i     = t_min + 2 * i * dt;                                      \
+            const float value_i = calculateL2NormAtValueOrder##N_minus_1(deriv, t_i);      \
+                                                                                           \
+            const float t_i_minus_1 = t_min + (2 * i - 1) * dt;                            \
+            const float value_i_minus_1 =                                                  \
+                calculateL2NormAtValueOrder##N_minus_1(deriv, t_i_minus_1);                \
+            sum_of_odd_s_values += value_i_minus_1;                                        \
+                                                                                           \
+            const float s_i = initial_value + 4 * sum_of_odd_s_values +                    \
+                              2 * sum_of_even_s_values + value_i;                          \
+                                                                                           \
+            parametrization.s_values[i] = s_i;                                             \
+            parametrization.t_values[i] = t_i;                                             \
+                                                                                           \
+            sum_of_even_s_values += value_i;                                               \
+        }                                                                                  \
+        for (size_t i = 0; i < parametrization.num_values; i++)                            \
+        {                                                                                  \
+            parametrization.s_values[i] *= dt / 3;                                         \
+        }                                                                                  \
+    }
+GENERATE_2D_POLYNOMIAL_GET_ARC_LENGTH_PARAMETRIZATION_FUNCTION_DEFINITION(3, 2)
+GENERATE_2D_POLYNOMIAL_GET_ARC_LENGTH_PARAMETRIZATION_FUNCTION_DEFINITION(2, 1)
+GENERATE_2D_POLYNOMIAL_GET_ARC_LENGTH_PARAMETRIZATION_FUNCTION_DEFINITION(1, 0)
+
+#define GENERATE_2D_POLYNOMIAL_GET_POSITION_AT_ARC_LENGTH_FUNCTION_DEFINITION(N)         \
+    Vector2d_t shared_polynomial2d_getPositionAtArcLengthOrder##N(                       \
```

Comment:
A comment for what `s` is would be great, or a more descriptive variable name

Reply:
Whoops, too much time in math land. Changed all instances of `s` to `arc_length`, here and everywhere else.